perfkitbenchmarker/providers/profitbricks/profitbricks_virtual_machine.py

```diff
@@ -0,0 +1,368 @@
+
+  Attributes:
+    cores: int. Number of CPU cores for a custom VM.
+    ram: int. Amount of RAM in MBs for a custom VM.
+  """
+
+  @classmethod
+  def _GetOptionDecoderConstructions(cls):
+    """Gets decoder classes and constructor args for each configurable option.
+
+    Returns:
+      dict. Maps option name string to a (ConfigOptionDecoder class, dict) pair.
+          The pair specifies a decoder class and its __init__() keyword
+          arguments to construct in order to decode the named option.
+    """
+    result = super(CustomMachineTypeSpec, cls)._GetOptionDecoderConstructions()
+    result.update({'cores': (option_decoders.IntDecoder, {'min': 1}),
+                   'ram': (option_decoders.IntDecoder, {'min': 1024})})
+    return result
+
+
+class MachineTypeDecoder(option_decoders.TypeVerifier):
+  """Decodes the machine_type option of a ProfitBricks VM config."""
+
+  def __init__(self, **kwargs):
+    super(MachineTypeDecoder, self).__init__((basestring, dict), **kwargs)
+
+  def Decode(self, value, component_full_name, flag_values):
+    """Decodes the machine_type option of a ProfitBricks VM config.
+
+    Args:
+      value: Either a string name of a PB machine type or a dict containing
+          'cpu' and 'memory' keys describing a custom VM.
+      component_full_name: string. Fully qualified name of the configurable
+          component containing the config option.
+      flag_values: flags.FlagValues. Runtime flag values to be propagated to
+          BaseSpec constructors.
+
+    Returns:
+      If value is a string, returns it unmodified. Otherwise, returns the
+      decoded CustomMachineTypeSpec.
+
+    Raises:
+      errors.Config.InvalidValue upon invalid input value.
+    """
+    super(MachineTypeDecoder, self).Decode(value, component_full_name,
+                                           flag_values)
+    if isinstance(value, basestring):
+      return value
+    return CustomMachineTypeSpec(self._GetOptionFullName(component_full_name),
+                                 flag_values=flag_values, **value)
+
+
+class ProfitBricksVmSpec(virtual_machine.BaseVmSpec):
+  """Object containing the information needed to create a
+  ProfitBricksVirtualMachine.
+
+  Attributes:
+    ram: None or int. RAM value in MB for custom ProfitBricks VM.
+    cores: None or int. CPU cores value for custom ProfitBricks VM.
+  """
+
+  CLOUD = providers.PROFITBRICKS
+
+  def __init__(self, *args, **kwargs):
+    super(ProfitBricksVmSpec, self).__init__(*args, **kwargs)
+    if isinstance(self.machine_type, CustomMachineTypeSpec):
+      logging.info('Using custom hardware configuration from config.')
+      self.cores = self.machine_type.cores
+      self.ram = self.machine_type.ram
+      self.machine_type = 'Custom (RAM: {}, Cores: {})'.format(self.ram,
+                                                               self.cores)
+    else:
+      try:
+          hardware = ast.literal_eval(self.machine_type)
```

Comment:
Adding an [_ApplyFlags](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/providers/gcp/gce_virtual_machine.py#L191) method will make it so that you don't need a separate case for custom machine types passed in through flags vs through configs since the config will reflect the flag value.